hathor/indexes/memory_address_index.py

```diff
@@ -30,12 +30,21 @@
 class MemoryAddressIndex(AddressIndex):
     """ Index of inputs/outputs by address
     """
+
+    index: DefaultDict[str, Set[bytes]]
+
     def __init__(self, pubsub: Optional['PubSubManager'] = None) -> None:
-        self.index: DefaultDict[str, Set[bytes]] = defaultdict(set)
         self.pubsub = pubsub
+        self.force_clear()
         if self.pubsub:
             self.subscribe_pubsub_events()
 
+    def get_db_name(self) -> Optional[str]:
```

Comment:
I see what you mean. I avoided doing anything that would make it easy to forget explicitly adding that method to non-memory indexes. Maybe if all memory indexes inherited from a `BaseMemoryIndex` instead of `BaseIndex`, that could be explicit enough.